### ISSUE TITLE
How to install the latest available version of puppet?

### ISSUE THREAD

**Issue by SebastianLopienski** [petems/vagrant-puppet-install#41]
I wanted `vagrant-puppet-install` to install the latest **available** version of puppet. So I put this line in `Vagrantfile`:

    config.puppet_install.puppet_version = :latest

It tries to install version 4.4.0 (as this is the latest version listed here: https://rubygems.org/gems/puppet/versions/). But this fails both on Fedora (`config.vm.box = "fedora/23-cloud-base"`) and on Ubuntu (`config.vm.box = "ubuntu/trusty64"`), because they have older versions of puppet in their default repos:

```
# Fedora
$ dnf list puppet 
puppet.noarch  4.2.1-2.fc23  fedora

# Ubuntu
$ apt list puppet
puppet/trusty-updates,now 3.4.3-1ubuntu1.1 all
```

The full error output on Fedora
```
sh install.sh -v 4.4.0 2>&1

Stdout from the command:

09:37:26 +0000 INFO: Version parameter defined: 4.4.0
09:37:26 +0000 INFO: Matching Puppet version to puppet-agent package version (See http://docs.puppetlabs.com/puppet/latest/reference/about_agent.html for more details)
09:37:26 +0000 CRIT: Unable to match requested puppet version to puppet-agent version - Check http://docs.puppetlabs.com/puppet/latest/reference/about_agent.html
09:37:26 +0000 CRIT: Please file a bug report at https://github.com/petems/puppet-install-shell/
09:37:26 +0000 CRIT: 
09:37:26 +0000 CRIT: Version: 4.4.0
09:37:26 +0000 CRIT: Platform: fedora
09:37:26 +0000 CRIT: Platform Version: 23
09:37:26 +0000 CRIT: Machine: x86_64
09:37:26 +0000 CRIT: OS: Linux
09:37:26 +0000 CRIT: 
09:37:26 +0000 CRIT: Please detail your operating system type, version and any other relevant details
```

How do I make `vagrant-puppet-install` to install the version of puppet that is currently available in package repositories?

**Comment by petems** [comment]
https://github.com/petems/puppet-install-shell/pull/45 should be fixed so it now gives 4.4 as the latest version.

**Comment by SebastianLopienski** [comment]
@petems Just to make sure that it was clear what I meant:

I would expect `vagrant-puppet-install` to install the latest puppet version as _available on a given platform_ (e.g. currently on Fedora 23: `yum install puppet` gives puppet version 4.2.1; on Ubuntu 14/Trusty: `apt-get install  puppet` gives puppet version 3.4.3); and _not_ the latest puppet version that exists (currently 4.4.0, but it doesn't exist in default repositories for Fedora and Ubuntu, so installation fails).

Is this possible somehow? Thanks for any hints.

**Comment by petems** [comment]
**tl;dr Long story, it's probably best to specify which version you want in the plugin to get consistent results.**

So long story here: https://docs.puppetlabs.com/puppet/latest/reference/about_agent.html

Basically, as of Puppet > 4, the package was renamed into `puppet-agent`, and bundled with all the various things it needs. This hasn't trickled down into platform specific repos, many of which are out of date and using EOL versions of Ruby, such as Ubuntu's 3.4.3. 

Fedora is a special case, because they actually built their own Puppet package for 4.0, still called `puppet`. 

So if you want to install Puppet > 4 on any platform other than Fedora, you have to use the puppet-agent install script from puppet-install-shell.

So for this plugin, it uses the puppet-install-shell script, which assumes if you want the latest version, that it will just run the standard script, and give you the latest 3.X version on your platform. 

I might change the plugin to default to use the puppet-agent installing script, but was going to wait until 3.X Puppet became maintenance mode, which is happening at the end of this year IIRC.

**Comment by SebastianLopienski** [comment]
OK... thanks for this clarification, @petems.

**Comment by petems** [comment]
No problem! It's a bit of a mess, as most package management problems are! :cry:

**Comment by petems** [comment]
@SebastianLopienski Can you try updating your vagrant-puppet-install plugin and trying again? The issue should be fixed with getting the latest version now :+1:

**Comment by SebastianLopienski** [comment]
@petems Just tried, works fine now. On both Fedora 23 and Ubuntu 14/Trusty, puppet 4.4.0 (puppet-agent 1.4.0) was correctly installed. Thanks!

**Comment by petems** [comment]
Awesome! :smile:
